### ISSUE TITLE
Request for Sample

### ISSUE THREAD

**Issue by BabyToy** [adrianplavka/typeswag#10]
I'm trying out typeswag and cannot get the parameters I needed into swagger.json.  
This is how I pass parameters in express:  
```typescript
import { Router } from "express";
const router = Router();

router.post("/processcountry", (request, response, next) => {
  const { country, actionId } = request.body;
  countries.process(country, actionId)
    .then((result) => {
      response.status(201).json({
        status: "OK",
        result
      });
    })
    .catch(next);
});
```

What is the equivalant in typeswag for ```const { country, actionId } = request.body```? I have tried this but I don't see the parameters in swagger.json:  
```typescript
class Processor {
  @Post("/processcountry")
  processCountry(@BodyParam("country") country: string, @BodyParam("actionId") actionId: string);
}
```

Thank you in advance.

**Comment by adrianplavka** [comment]
Hi, thanks for submitting an issue!

Just a quick question - as I've understood it from your example, are you using `typeswag` on your own or are you using it with some web-framework (`routing-controllers`)? If you're using it on your own, it won't work, as you need another library, that handles the HTTP for you! 

The decorator `@BodyParam` was, indeed, a correct way to get a body property, but it a relic from `tsoa`'s library, which handled it.

It may not be that obvious, but this library is still not complete (as you can see form the milestones in the README, there are a couple of things missing in order to fully integrate this into other frameworks, that use decorators).

If I can make a suggestion, you should look into `tsoa` library. It handles both of the things fairly well and may be better suited for your needs.

Also if you have any more questions, just drop them here!

**Comment by BabyToy** [comment]
Hi Adrian,  

Thank you for the response, and yes, I am using routing-controller together with Express.  

After 4 days for pulling my hair I was able to make a swagger.json with okay-ish results (snipped):  
```    "/processcountry": {
      "post": {
        "operationId": "ProcessCountry",
        "produces": [
          "application/json"
        ],
        "responses": {
          "200": {
            "description": "Ok",
            "schema": {
              "type": "string"
            }
          }
        },
        "security": [],
        "parameters": [{
            "in": "body",
            "name": "body",
            "schema": {
              "properties": {
                "country": {
                  "type": "string"
                },
                "actionId": {
                  "type": "string"
                }
              },
              "title": "ProcessCountryBody",
              "type": "object",
              "required": [
                "country",
                "actionId"
              ]
            }
          }
        ]
      }
    }
```
This is based on my path definition in the contoller:  
```  @Post("/processcountry")
  @HttpCode(201)
  processCountry(
    @BodyProp("country")
    @BodyParam("country")
    country: string,
    @BodyProp("actionId")
    @BodyParam("actionId")
    actionId: string
  ) {
    winston.info(`country: ${country} actionId: ${actionId}`);
    return "Country processed";
  }
```
Not sure if this is correct - please correct me if I'm wrong. What happened here is that for each ```@BodyParam``` there is a ```@BodyProp```. Looks weird, I know, but it works and I would like your opinion if this is the correct approach.  

The HTTP response code is supposed to be 201 but is 200 in the swagger file. I guess I need to do more diving in routing controller.

Thank you  and regards,

IAN

**Comment by adrianplavka** [comment]
Yes, your code is correct, but as you can see, there is a lot of redundancy, specifying body properties twice is not what was planned initially with this library, but rather allow users to map & register framework's decorators, so the Swagger generator knows where to look!

`@HttpCode` is also an alias to library's `@SuccessResponse`, so when you add it above, you will get your correct 201 code.

What needs to be done is to create an API for these custom method & parameter decorators (that will look something like the `registerRouteDecorator` in the README example). It is still not done, as this library was initially created for our internal project, but with the idea of this customizabillity.

**Comment by BabyToy** [comment]
I replaced ```@SuccessResponse``` with ```@HttpCode``` now swagger.json says:
```        "responses": {
          "200": {
            "description": "Ok",
            "schema": {
              "type": "string"
            }
          }
```
previously it was 201.

I commented out ```@BodyProp``` and now ```parameters``` becomes:
```
        "parameters": []
```
However, the parameters still have values when tested on Postman.
